community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/planner/logical/steps/verifyBestPlan.scala

```diff
@@ -39,10 +39,15 @@ object verifyBestPlan extends PlanTransformer[PlannerQuery] {
         val b: PlannerQuery = constructed.withoutHints(constructed.allHints)
         if (a != b) {
           // unknown planner issue failed to find plan (without regard for differences in hints)
-          throw new CantHandleQueryException(s"Expected \n$expected \n\n\nInstead, got: \n$constructed")
+          throw new InvalidArgumentException(s"Expected \n$expected \n\n\nInstead, got: \n$constructed")
         } else {
           // unknown planner issue failed to find plan matching hints (i.e. "implicit hints")
-          throw new CantHandleQueryException(s"Expected \n${expected.allHints} \n\n\nInstead, got: \n${constructed.allHints}")
+          val expectedHints = expected.allHints.mkString(System.lineSeparator())
+          val actualHints = constructed.allHints.mkString(System.lineSeparator())
+          val message =
+            s"Something went when trying to fulfill the index hints of your query.\n" +
```

Comment:
_Something went ..._ where did it go? I think you meant to say _Something went *wrong*..._, I would actually prefer a slightly different wording that doesn't sound as much like we don't know what is going on.

Furthermore we have hints that are not index hints, is this part of the code only dealing with index hints as the message says, or should the message change to talk about hints in general. (I'm thinking of how a scan hint is not technically an index hint, and can be used in the same place).

How about `Failed to fulfil the hints of the query.`?

I'm also wondering if we instead should be using a more specific exception type, rather than encoding all of this information in a message only.